### ISSUE TITLE
Improve API for SPARQL SELECT query results

### ISSUE THREAD

**Issue by Aklakan** [SANSA-Stack/SANSA-Query#47]
This issue combines the related issues #13 and #43 into a single one.

**Comment by LorenzBuehmann** [comment]
The naming issue was also reported here: https://stackoverflow.com/questions/62118243/sparql-column-headings-with-different-name

**Comment by LorenzBuehmann** [comment]
Will be fixed with release 0.8.0 when switching to Ontop engine as SPARQL-to-SQL rewriter as default

**Comment by shredder47** [comment]
any news on this ?

**Comment by JNKHunter** [comment]
Here is a temporary solution that works for my purposes that returns a dataframe with the expected column names. By decomposing rdd.sparql, the rewrite object can be used to obtain the column mappings: 
https://gist.github.com/JNKHunter/c16caa882993facb31a273ec274cb8e3

Warning: Sansa query often returns more than one column for each sparql selected element. In those cases this code simply concatenates those columns since columns after the first tend to contain empty strings. Concatenation may not be what you want, however I've yet to discover what those empty string dataframe columns represent.

**Comment by LorenzBuehmann** [comment]
Hi, sorry for the delay. Release is almost there.

Just to recap, with 0.8.0 we support another SPARQL-to-SQL rewriter based on Ontop. Given that RDF allows for mixed content for a single property we also prefer to return an RDD of bindings now to be more flexible. 
As you already recognized, generated SQL queries can return multiple columns for the same property because of having the data partitioned into different tables for each datatype of objects. So the result of a SQL query is basically some kind of raw result which needs for some post-processing step afterwards.
That is what @JNKHunter did in his workaround as far as I can see.

In theory, one could create a DataFrame back from the RDD of bindings, but this isn't implemented yet - might also be less efficient for sure. We might add it as an alternative to a follow up release.

So, basically, this issue is indirectly resolved by using the RDD of bindings way for the result, though not yet for the DataFrame result. Thus, I'll keep this issue open, but change the version to 0.9.0 milestone.
